### PR TITLE
Support sessions saving by metad

### PR DESCRIPTION
**Pull request by laura-ding**

### CODE REVIEW

File: src/common/interface/common.thrift [hunk 1]
```diff
@@ -180,3 +184,97 @@ struct KeyValue {
...
+    kLimit = 71,
+    kGroupBy = 72,
+    kReturn = 73,
+    kCreateSnapshot = 74,
+    kDropSnapshot = 75,
+    kAdminJob = 76,
+    kGetSubgraph = 77,
+    kShowSessions = 78,
+    kGetSession = 79,
+} (cpp.enum_strict cpp.type = "nebula::SentenceKind")
```
Comment (yixinglu) [inline src/common/interface/common.thrift:270]: I feel that it's not a good practice to move the graph sentence kind here.

Comment (laura-ding) [reply src/common/interface/common.thrift:270]: the session will save the sentence kind. Do you have a better idea?

Comment (CPWstatic) [reply src/common/interface/common.thrift:270]: We could use string instead of enum.

Comment (laura-ding) [reply src/common/interface/common.thrift:270]: @CPWstatic string takes up storage.

File: src/common/datatypes/Session.h [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+struct Session {
+    SessionID sessionId;
+    Timestamp createTime;
+    Timestamp updateTime;
+    std::string userName;
+    std::string spaceName;
+    HostAddr graphAddr;
+    int32_t timezone{0};
+    std::unordered_map<SentenceKind, int64_t> cmdStats;
```
Comment (darionyaphet) [inline src/common/datatypes/Session.h:107]: Seems  the session is a logic connection between user and nebula 

Do we need support cmdStats to specified the sentence ?

File: src/common/interface/meta.thrift [hunk 3]
```diff
@@ -981,6 +981,57 @@ struct GetStatisResp {
...
+}
+
+struct GetSessionReq {
+    1: common.SessionID  session_id,
+}
+
+struct GetSessionResp {
+    1: ErrorCode            code,
+    2: common.HostAddr      leader,
+    3: Session              session,
```
Comment (dutor) [inline src/common/interface/meta.thrift:1028]: Should fields like this be optional?

**Review by dutor — COMMENTED** [review]
Well done.

File: src/common/interface/meta.thrift (hunk 3 above)
Comment (laura-ding) [reply src/common/interface/meta.thrift:1028]: Why? when the error?

**Review by dutor — DISMISSED** [review]
LGTM.

File: src/common/interface/graph.thrift [hunk 4]
```diff
@@ -55,6 +55,9 @@ enum ErrorCode {
 
     // semantic error
     E_SEMANTIC_ERROR = -12,
+
+    // Exceeding the maximum number of connections
+    E_OUT_OF_MAX_CONNECTIONS = -13,
```
Comment (dutor) [inline src/common/interface/graph.thrift:60]: Better rename it to `E_TOO_MANY_CONNECTIONS`

**Review by dutor — DISMISSED** [review]
LGTM.

**Review by dutor — DISMISSED** [review]

File: src/common/interface/meta.thrift [hunk 5]
```diff
@@ -1020,6 +1020,57 @@ struct ListFTClientsResp {
...
 
+struct Session {
+    1: common.SessionID session_id,
+    2: common.Timestamp create_time,
+    3: common.Timestamp update_time,
+    4: binary user_name,
+    5: binary space_name,
+    6: common.HostAddr graph_addr,
+    7: i32 timezone,
+    8: binary client_ip,
```
Comment (Shylock-Hg) [inline src/common/interface/meta.thrift:1031]: i32 for ip?

File: src/common/interface/meta.thrift [hunk 6]
```diff
@@ -1020,6 +1020,57 @@ struct ListFTClientsResp {
...
+    9: map<binary, common.Value> configs,
```
Comment (Shylock-Hg) [inline src/common/interface/meta.thrift:1032]: unordered_map

File: src/common/interface/meta.thrift (hunk 5 above)
Comment (laura-ding) [reply src/common/interface/meta.thrift:1031]: the same as HostAddr

File: src/common/interface/meta.thrift (hunk 6 above)
Comment (laura-ding) [reply src/common/interface/meta.thrift:1032]: good catch

**Review by dutor — APPROVED** [review]
LGTM.

**Review by darionyaphet — APPROVED** [review]

**Pull request merged**